signals/signal.go

```diff
@@ -41,3 +43,33 @@ func SetupSignalHandler() (stopCh <-chan struct{}) {
 
 	return stop
 }
+
+// NewContext creates a new context with SetupSignalHandler()
+// as our Done() channel.
+func NewContext() context.Context {
+	return &signalContext{stopCh: SetupSignalHandler()}
+}
+
+type signalContext struct {
+	stopCh <-chan struct{}
+}
+
+// Deadline implements context.Context
+func (scc *signalContext) Deadline() (deadline time.Time, ok bool) {
+	return time.Now(), false
```

Comment:
I don't think we need to return `time.Now()`

```suggestion
	return
```

Reply:
Done